text/0001-cdk-update.md

```diff
@@ -0,0 +1,299 @@
+Assets will be uploaded to their storage medium. The CDK will then use AWS SDK
+commands to directly modify the stack resources in-place to synchronize them
+with your local code. The accelerate command can act on one or more stacks in
+the application.
+
+#### Usage
+
+To run a one-shot update, you can invoke the `cdk deploy --accelerated` command
+on some or all of your stacks:
+
+```
+$ cdk deploy --accelerated ApplicationStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[AppFunction]: File asset changed
+ApplicationStack can be updated
+1/1 stacks can be updated:
+Updating LambdaFunction in ApplicationStack
+...
+Update complete!
+```
+
+If the update is to an attribute of the stack that cannot be updated, the
+command will offer to perform a full deployment:
+
+```
+$ cdk deploy --accelerated ApplicationStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[Appfunction]: Function data changed
+ApplicationStack has changes that can not be rapidly updated.
+Perform a full deployment of ApplicationStack? [Y/n]: Y
+Deploying ApplicationStack
+...
+Done!
+```
+
+When multiple stacks are provided, update only occurs if all pending changes are
+suitable for update. Otherwise, a full deployment is done:
+
+```
+$ cdk deploy --accelerated ApplicationStack DataStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[Appfunction]: Function data changed
+ApplicationStack has changes that can not be rapidly updated.
+Checking stack DataStack for possible accelerated update
+ * LambdaFunction[StreamFunction]: File asset changed
+DataStack can be updated
+1/2 stacks can be updated: Perform a full deployment? [Y/n]? Y
+Deploying ApplicationStack
+...
+Deploying DataStack
+...
+Done!
+```
+
+In addition to running in a one shot mode, the `cdk deploy --accelerated`
+command also has a `--watch` command line option that enable it to monitor the
+assets on disk and perform an update when they change.
+
+```
+$ cdk deploy --accelerated --watch ApplicationStack DataStack
+Checking stack ApplicationStack for possible accelerated update
+ * LambdaFunction[Appfunction]: No Changes
+ApplicationStack can *not* be updated
+Checking stack DataStack for possible accelerated update
+ * LambdaFunction[StreamFunction]: No Changes
+
+Watching stack inputs for changes:
+ * LambdaFunction[Appfunction]: <spinner>
+ * LambdaFunction[StreamFunction]: <spinner>
+
+```
+
+The watcher can only watch for file changes. For Lambda code, that means that
+directories will be watched for changes and zipped if they change. If the file
+asset is a zip file, then the update will fire whenever that changes. The
+existing docker asset builder will be used to watch for changes in local docker
+images.
+
+#### Resource Support
+
+- AWS Lambda `Function`
+  - file and directory assets
+- AWS Fargate
+  - image assets
+- ECS
+  - image assets
+
+#### Future Support Plans
+
+- StepFunctions
```

Comment:
Might be out of scope but it may be worth considering partial synthesis too. I'm not sure if other constructs could benefit from this besides state machines.

When writing [StateMachines](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.StateMachine.html) in CDK, you lose the workflow visualization and it's harder to understand the flow without seeing the graph. The only way to see it is by deploying the stack and viewing the created resource in the console. AWS Toolkit for Visual Studio Code supports state machine visualization from JSON or YAML which can be generated from `cdk synth`. But we don't need the full CFN template or reference resolution, just the state machine structure. Being able to view the graph even before deploying to get the JSON definition would shorten the dev cycle even further.

Reply:
That's a good suggestion @wong-a, but this is tricky. It's really difficult to perform partial synthesis in CDK, because the various constructs can reference each other, including Stacks. So, performing partial synthesis can result in very weird errors because of generating incorrect CloudFormation templates.